[Bug] [ST Engine] GetTaskGroupMetricsOperation error

### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-seatunnel/issues?q=is%3Aissue+label%3A%22bug%22) and found no similar issues.


### What happened

Hint: You can enable the logging of stack traces with the following system property: -Dhazelcast.slow.operation.detector.stacktrace.logging.enabled
2022-12-03 10:08:11,235 ERROR com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl - [localhost]:5802 [gaojun_ClusterFaultToleranceIT_testBatchJobRunOkIn3Node] [5.1] Problem while reading DataSerializable, namespace: 0, ID: 0, class: 'org.apache.seatunnel.engine.server.operation.GetTaskGroupMetricsOperation', exception: Classes conforming to DataSerializable should provide a no-arguments constructor.
com.hazelcast.nio.serialization.HazelcastSerializationException: Problem while reading DataSerializable, namespace: 0, ID: 0, class: 'org.apache.seatunnel.engine.server.operation.GetTaskGroupMetricsOperation', exception: Classes conforming to DataSerializable should provide a no-arguments constructor.
	at com.hazelcast.internal.serialization.impl.DataSerializableSerializer.rethrowReadException(DataSerializableSerializer.java:183) ~[hazelcast-5.1.jar:5.1]
	at com.hazelcast.internal.serialization.impl.DataSerializableSerializer.readInternal(DataSerializableSerializer.java:164) ~[hazelcast-5.1.jar:5.1]
	at com.hazelcast.internal.serialization.impl.DataSerializableSerializer.read(DataSerializableSerializer.java:106) ~[hazelcast-5.1.jar:5.1]
	at com.hazelcast.internal.serialization.impl.DataSerializableSerializer.read(DataSerializableSerializer.java:51) ~[hazelcast-5.1.jar:5.1]
	at com.hazelcast.internal.serialization.impl.StreamSerializerAdapter.read(StreamSerializerAdapter.java:44) ~[hazelcast-5.1.jar:5.1]
	at com.hazelcast.internal.serialization.impl.AbstractSerializationService.toObject(AbstractSerializationService.java:268) ~[hazelcast-5.1.jar:5.1]
	at com.hazelcast.spi.impl.NodeEngineImpl.toObject(NodeEngineImpl.java:385) ~[hazelcast-5.1.jar:5.1]
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:456) [hazelcast-5.1.jar:5.1]
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.process(OperationThread.java:197) [hazelcast-5.1.jar:5.1]
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.process(OperationThread.java:137) [hazelcast-5.1.jar:5.1]
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.executeRun(OperationThread.java:123) [hazelcast-5.1.jar:5.1]
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:102) [hazelcast-5.1.jar:5.1]
Caused by: java.lang.NoSuchMethodException: Classes conforming to DataSerializable should provide a no-arguments constructor.
	at com.hazelcast.internal.serialization.impl.DataSerializableSerializer.tryClarifyNoSuchMethodException(DataSerializableSerializer.java:221) ~[hazelcast-5.1.jar:5.1]
	at com.hazelcast.internal.serialization.impl.DataSerializableSerializer.readInternal(DataSerializableSerializer.java:163) ~[hazelcast-5.1.jar:5.1]
	... 10 more
Caused by: java.lang.NoSuchMethodException: org.apache.seatunnel.engine.server.operation.GetTaskGroupMetricsOperation.<init>()
	at java.lang.Class.getConstructor0(Class.java:3082) ~[?:1.8.0_181]
	at java.lang.Class.getDeclaredConstructor(Class.java:2178) ~[?:1.8.0_181]
	at com.hazelcast.internal.nio.ClassLoaderUtil.newInstance(ClassLoaderUtil.java:222) ~[hazelcast-5.1.jar:5.1]
	at com.hazelcast.internal.nio.ClassLoaderUtil.newInstance0(ClassLoaderUtil.java:217) ~[hazelcast-5.1.jar:5.1]
	at com.hazelcast.internal.nio.ClassLoaderUtil.newInstance(ClassLoaderUtil.java:197) ~[hazelcast-5.1.jar:5.1]
	at com.hazelcast.internal.serialization.impl.DataSerializableSerializer.readInternal(DataSerializableSerializer.java:152) ~[hazelcast-5.1.jar:5.1]
	... 10 more

### SeaTunnel Version

dev

### SeaTunnel Config

```conf
Run ClusterFaultToleranceIT#testBatchJobRunOkIn3Node test case.
```


### Running Command

```shell
Run ClusterFaultToleranceIT#testBatchJobRunOkIn3Node test case.
```


### Error Exception

```log
""
```


### Flink or Spark Version

_No response_

### Java or Scala Version

_No response_

### Screenshots

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

--- a/seatunnel-api/src/main/java/org/apache/seatunnel/api/common/metrics/RawJobMetrics.java
+++ b/seatunnel-api/src/main/java/org/apache/seatunnel/api/common/metrics/RawJobMetrics.java
@@ -17,9 +17,10 @@
 
 package org.apache.seatunnel.api.common.metrics;
 
+import java.io.Serializable;
 import java.util.Arrays;
 
-public final class RawJobMetrics {
+public final class RawJobMetrics implements Serializable {
 
     private long timestamp;
     private byte[] blob;

--- a/seatunnel-api/src/main/java/org/apache/seatunnel/api/env/EnvCommonOptions.java
+++ b/seatunnel-api/src/main/java/org/apache/seatunnel/api/env/EnvCommonOptions.java
@@ -34,7 +34,7 @@ public class EnvCommonOptions {
     public static final Option<String> JOB_NAME =
         Options.key("job.name")
             .stringType()
-            .noDefaultValue()
+            .defaultValue("SeaTunnel_Job")
             .withDescription("The job name of this job");
 
     public static final Option<JobMode> JOB_MODE =

--- a/seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/java/org/apache/seatunnel/engine/e2e/ClusterFaultToleranceTwoPipelineIT.java
+++ b/seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/java/org/apache/seatunnel/engine/e2e/ClusterFaultToleranceTwoPipelineIT.java
@@ -38,6 +38,7 @@ import lombok.NonNull;
 import lombok.extern.slf4j.Slf4j;
 import org.awaitility.Awaitility;
 import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Disabled;
 import org.junit.jupiter.api.Test;
 import org.testcontainers.shaded.org.apache.commons.lang3.tuple.ImmutablePair;
 
@@ -49,10 +50,12 @@ import java.util.concurrent.ExecutionException;
 import java.util.concurrent.TimeUnit;
 
 /**
- * Cluster fault tolerance test. Test the job recovery capability and data consistency assurance capability in case of cluster node failure
+ * Cluster fault tolerance test. Test the job which have two pipelines can recovery capability and data consistency assurance capability in case of cluster node failure
  */
 @Slf4j
-public class ClusterFaultToleranceIT {
+public class ClusterFaultToleranceTwoPipelineIT {
+
+    public static final String TEST_TEMPLATE_FILE_NAME = "cluster_batch_fake_to_localfile_two_pipeline_template.conf";
 
     public static final String DYNAMIC_TEST_CASE_NAME = "dynamic_test_case_name";
 
@@ -64,9 +67,10 @@ public class ClusterFaultToleranceIT {
 
     @SuppressWarnings("checkstyle:RegexpSingleline")
     @Test
+    @Disabled("disabled until we fix the file sink filename duplicate bug when use SeaTunnel Engine")
     public void testBatchJobRunOkIn3Node() throws ExecutionException, InterruptedException {
         String testCaseName = "testBatchJobRunOkIn3Node";
-        String testClusterName = "ClusterFaultToleranceIT_testBatchJobRunOkIn3Node";
+        String testClusterName = "ClusterFaultToleranceTwoPipelineIT_testBatchJobRunOkIn3Node";
         long testRowNumber = 1000;
         int testParallelism = 6;
 
@@ -92,7 +96,7 @@ public class ClusterFaultToleranceIT {
 
             Common.setDeployMode(DeployMode.CLIENT);
             ImmutablePair<String, String> testResources =
-                createTestResources(testCaseName, JobMode.BATCH, testRowNumber, testParallelism);
+                createTestResources(testCaseName, JobMode.BATCH, testRowNumber, testParallelism, TEST_TEMPLATE_FILE_NAME);
             JobConfig jobConfig = new JobConfig();
             jobConfig.setName(testCaseName);
 
@@ -117,8 +121,7 @@ public class ClusterFaultToleranceIT {
                 });
 
             Long fileLineNumberFromDir = FileUtils.getFileLineNumberFromDir(testResources.getLeft());
-            Assertions.assertEquals(testRowNumber * testParallelism, fileLineNumberFromDir);
-            System.out.println(engineClient.getJobMetrics(clientJobProxy.getJobId()));
+            Assertions.assertEquals(testRowNumber * testParallelism * 2, fileLineNumberFromDir);
         } finally {
             if (engineClient != null) {
                 engineClient.shutdown();
@@ -147,8 +150,11 @@ public class ClusterFaultToleranceIT {
      * @param rowNumber    row.num per FakeSource parallelism
      * @param parallelism  FakeSource parallelism
      */
-    private ImmutablePair<String, String> createTestResources(@NonNull String testCaseName, @NonNull JobMode jobMode,
-                                                              long rowNumber, int parallelism) {
+    private ImmutablePair<String, String> createTestResources(@NonNull String testCaseName,
+                                                              @NonNull JobMode jobMode,
+                                                              long rowNumber,
+                                                              int parallelism,
+                                                              @NonNull String templateFileName) {
         checkArgument(rowNumber > 0, "rowNumber must greater than 0");
         checkArgument(parallelism > 0, "parallelism must greater than 0");
         Map<String, String> valueMap = new HashMap<>();
@@ -166,17 +172,17 @@ public class ClusterFaultToleranceIT {
         String targetConfigFilePath =
             File.separator + "tmp" + File.separator + "test_conf" + File.separator + testCaseName +
                 ".conf";
-        TestUtils.createTestConfigFileFromTemplate("cluster_batch_fake_to_localfile_template.conf", valueMap,
-            targetConfigFilePath);
+        TestUtils.createTestConfigFileFromTemplate(templateFileName, valueMap, targetConfigFilePath);
 
         return new ImmutablePair<>(targetDir, targetConfigFilePath);
     }
 
     @SuppressWarnings("checkstyle:RegexpSingleline")
     @Test
+    @Disabled("disabled until we fix the file sink filename duplicate bug when use SeaTunnel Engine")
     public void testStreamJobRunOkIn3Node() throws ExecutionException, InterruptedException {
         String testCaseName = "testStreamJobRunOkIn3Node";
-        String testClusterName = "ClusterFaultToleranceIT_testStreamJobRunOkIn3Node";
+        String testClusterName = "ClusterFaultToleranceTwoPipelineIT_testStreamJobRunOkIn3Node";
         long testRowNumber = 1000;
         int testParallelism = 6;
         HazelcastInstanceImpl node1 = null;
@@ -200,7 +206,7 @@ public class ClusterFaultToleranceIT {
 
             Common.setDeployMode(DeployMode.CLIENT);
             ImmutablePair<String, String> testResources =
-                createTestResources(testCaseName, JobMode.STREAMING, testRowNumber, testParallelism);
+                createTestResources(testCaseName, JobMode.STREAMING, testRowNumber, testParallelism, TEST_TEMPLATE_FILE_NAME);
             JobConfig jobConfig = new JobConfig();
             jobConfig.setName(testCaseName);
 
@@ -221,7 +227,7 @@ public class ClusterFaultToleranceIT {
                     Thread.sleep(2000);
                     System.out.println(FileUtils.getFileLineNumberFromDir(testResources.getLeft()));
                     Assertions.assertTrue(JobStatus.RUNNING.equals(clientJobProxy.getJobStatus()) &&
-                        testRowNumber * testParallelism == FileUtils.getFileLineNumberFromDir(testResources.getLeft()));
+                        testRowNumber * testParallelism * 2 == FileUtils.getFileLineNumberFromDir(testResources.getLeft()));
                 });
 
             clientJobProxy.cancelJob();
@@ -231,7 +237,7 @@ public class ClusterFaultToleranceIT {
                     objectCompletableFuture.isDone() && JobStatus.CANCELED.equals(objectCompletableFuture.get())));
 
             Long fileLineNumberFromDir = FileUtils.getFileLineNumberFromDir(testResources.getLeft());
-            Assertions.assertEquals(testRowNumber * testParallelism, fileLineNumberFromDir);
+            Assertions.assertEquals(testRowNumber * testParallelism * 2, fileLineNumberFromDir);
 
         } finally {
             if (engineClient != null) {
@@ -254,9 +260,10 @@ public class ClusterFaultToleranceIT {
 
     @SuppressWarnings("checkstyle:RegexpSingleline")
     @Test
+    @Disabled("disabled until we fix the file sink filename duplicate bug when use SeaTunnel Engine")
     public void testBatchJobRestoreIn3NodeWorkerDown() throws ExecutionException, InterruptedException {
         String testCaseName = "testBatchJobRestoreIn3NodeWorkerDown";
-        String testClusterName = "ClusterFaultToleranceIT_testBatchJobRestoreIn3NodeWorkerDown";
+        String testClusterName = "ClusterFaultToleranceTwoPipelineIT_testBatchJobRestoreIn3NodeWorkerDown";
         long testRowNumber = 1000;
         int testParallelism = 2;
         HazelcastInstanceImpl node1 = null;
@@ -280,7 +287,7 @@ public class ClusterFaultToleranceIT {
 
             Common.setDeployMode(DeployMode.CLIENT);
             ImmutablePair<String, String> testResources =
-                createTestResources(testCaseName, JobMode.BATCH, testRowNumber, testParallelism);
+                createTestResources(testCaseName, JobMode.BATCH, testRowNumber, testParallelism, TEST_TEMPLATE_FILE_NAME);
             JobConfig jobConfig = new JobConfig();
             jobConfig.setName(testCaseName);
 
@@ -313,7 +320,7 @@ public class ClusterFaultToleranceIT {
                     objectCompletableFuture.isDone() && JobStatus.FINISHED.equals(objectCompletableFuture.get())));
 
             Long fileLineNumberFromDir = FileUtils.getFileLineNumberFromDir(testResources.getLeft());
-            Assertions.assertEquals(testRowNumber * testParallelism, fileLineNumberFromDir);
+            Assertions.assertEquals(testRowNumber * testParallelism * 2, fileLineNumberFromDir);
 
         } finally {
             if (engineClient != null) {
@@ -336,9 +343,10 @@ public class ClusterFaultToleranceIT {
 
     @SuppressWarnings("checkstyle:RegexpSingleline")
     @Test
+    @Disabled("disabled until we fix the file sink filename duplicate bug when use SeaTunnel Engine")
     public void testStreamJobRestoreIn3NodeWorkerDown() throws ExecutionException, InterruptedException {
         String testCaseName = "testStreamJobRestoreIn3NodeWorkerDown";
-        String testClusterName = "ClusterFaultToleranceIT_testStreamJobRestoreIn3NodeWorkerDown";
+        String testClusterName = "ClusterFaultToleranceTwoPipelineIT_testStreamJobRestoreIn3NodeWorkerDown";
         long testRowNumber = 1000;
         int testParallelism = 6;
         HazelcastInstanceImpl node1 = null;
@@ -362,7 +370,7 @@ public class ClusterFaultToleranceIT {
 
             Common.setDeployMode(DeployMode.CLIENT);
             ImmutablePair<String, String> testResources =
-                createTestResources(testCaseName, JobMode.STREAMING, testRowNumber, testParallelism);
+                createTestResources(testCaseName, JobMode.STREAMING, testRowNumber, testParallelism, TEST_TEMPLATE_FILE_NAME);
             JobConfig jobConfig = new JobConfig();
             jobConfig.setName(testCaseName);
 
@@ -397,7 +405,7 @@ public class ClusterFaultToleranceIT {
                     Thread.sleep(2000);
                     System.out.println(FileUtils.getFileLineNumberFromDir(testResources.getLeft()));
                     Assertions.assertTrue(JobStatus.RUNNING.equals(clientJobProxy.getJobStatus()) &&
-                        testRowNumber * testParallelism == FileUtils.getFileLineNumberFromDir(testResources.getLeft()));
+                        testRowNumber * testParallelism * 2 == FileUtils.getFileLineNumberFromDir(testResources.getLeft()));
                 });
 
             // sleep 10s and expect the job don't write more rows.
@@ -410,7 +418,7 @@ public class ClusterFaultToleranceIT {
 
             // check the final rows
             Long fileLineNumberFromDir = FileUtils.getFileLineNumberFromDir(testResources.getLeft());
-            Assertions.assertEquals(testRowNumber * testParallelism, fileLineNumberFromDir);
+            Assertions.assertEquals(testRowNumber * testParallelism * 2, fileLineNumberFromDir);
 
         } finally {
             if (engineClient != null) {
@@ -433,9 +441,10 @@ public class ClusterFaultToleranceIT {
 
     @SuppressWarnings("checkstyle:RegexpSingleline")
     @Test
+    @Disabled("disabled until we fix the file sink filename duplicate bug when use SeaTunnel Engine")
     public void testBatchJobRestoreIn3NodeMasterDown() throws ExecutionException, InterruptedException {
         String testCaseName = "testBatchJobRestoreIn3NodeMasterDown";
-        String testClusterName = "ClusterFaultToleranceIT_testBatchJobRestoreIn3NodeMasterDown";
+        String testClusterName = "ClusterFaultToleranceTwoPipelineIT_testBatchJobRestoreIn3NodeMasterDown";
         long testRowNumber = 1000;
         int testParallelism = 6;
         HazelcastInstanceImpl node1 = null;
@@ -459,7 +468,7 @@ public class ClusterFaultToleranceIT {
 
             Common.setDeployMode(DeployMode.CLIENT);
             ImmutablePair<String, String> testResources =
-                createTestResources(testCaseName, JobMode.BATCH, testRowNumber, testParallelism);
+                createTestResources(testCaseName, JobMode.BATCH, testRowNumber, testParallelism, TEST_TEMPLATE_FILE_NAME);
             JobConfig jobConfig = new JobConfig();
             jobConfig.setName(testCaseName);
 
@@ -492,7 +501,7 @@ public class ClusterFaultToleranceIT {
                     objectCompletableFuture.isDone() && JobStatus.FINISHED.equals(objectCompletableFuture.get())));
 
             Long fileLineNumberFromDir = FileUtils.getFileLineNumberFromDir(testResources.getLeft());
-            Assertions.assertEquals(testRowNumber * testParallelism, fileLineNumberFromDir);
+            Assertions.assertEquals(testRowNumber * testParallelism * 2, fileLineNumberFromDir);
 
         } finally {
             if (engineClient != null) {
@@ -515,9 +524,10 @@ public class ClusterFaultToleranceIT {
 
     @SuppressWarnings("checkstyle:RegexpSingleline")
     @Test
+    @Disabled("disabled until we fix the file sink filename duplicate bug when use SeaTunnel Engine")
     public void testStreamJobRestoreIn3NodeMasterDown() throws ExecutionException, InterruptedException {
         String testCaseName = "testStreamJobRestoreIn3NodeMasterDown";
-        String testClusterName = "ClusterFaultToleranceIT_testStreamJobRestoreIn3NodeMasterDown";
+        String testClusterName = "ClusterFaultToleranceTwoPipelineIT_testStreamJobRestoreIn3NodeMasterDown";
         long testRowNumber = 1000;
         int testParallelism = 6;
         HazelcastInstanceImpl node1 = null;
@@ -541,7 +551,7 @@ public class ClusterFaultToleranceIT {
 
             Common.setDeployMode(DeployMode.CLIENT);
             ImmutablePair<String, String> testResources =
-                createTestResources(testCaseName, JobMode.STREAMING, testRowNumber, testParallelism);
+                createTestResources(testCaseName, JobMode.STREAMING, testRowNumber, testParallelism, TEST_TEMPLATE_FILE_NAME);
             JobConfig jobConfig = new JobConfig();
             jobConfig.setName(testCaseName);
 
@@ -575,7 +585,7 @@ public class ClusterFaultToleranceIT {
                     Thread.sleep(2000);
                     System.out.println(FileUtils.getFileLineNumberFromDir(testResources.getLeft()));
                     Assertions.assertTrue(JobStatus.RUNNING.equals(clientJobProxy.getJobStatus()) &&
-                        testRowNumber * testParallelism == FileUtils.getFileLineNumberFromDir(testResources.getLeft()));
+                        testRowNumber * testParallelism * 2 == FileUtils.getFileLineNumberFromDir(testResources.getLeft()));
                 });
 
             // sleep 10s and expect the job don't write more rows.
@@ -588,7 +598,7 @@ public class ClusterFaultToleranceIT {
 
             // check the final rows
             Long fileLineNumberFromDir = FileUtils.getFileLineNumberFromDir(testResources.getLeft());
-            Assertions.assertEquals(testRowNumber * testParallelism, fileLineNumberFromDir);
+            Assertions.assertEquals(testRowNumber * testParallelism * 2, fileLineNumberFromDir);
 
         } finally {
             if (engineClient != null) {

--- a/seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/resources/cluster_batch_fake_to_localfile_two_pipeline_template.conf
+++ b/seatunnel-e2e/seatunnel-engine-e2e/connector-seatunnel-e2e-base/src/test/resources/cluster_batch_fake_to_localfile_two_pipeline_template.conf
@@ -1,0 +1,154 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+######
+###### This config file is a demonstration of streaming processing in seatunnel config
+######
+
+env {
+  # You can set flink configuration here
+  job.mode = "${dynamic_job_mode}" # dynamic_job_mode will be replace to the final file name before test run
+  checkpoint.interval = 5000
+  #execution.checkpoint.data-uri = "hdfs://localhost:9000/checkpoint"
+}
+
+source {
+  # This is a example source plugin **only for test and demonstrate the feature source plugin**
+    FakeSource {
+      result_table_name = table1
+      row.num = ${dynamic_test_row_num_per_parallelism}
+      split.num = 5
+      split.read-interval = 3000
+      map.size = 10
+      array.size = 10
+      bytes.length = 10
+      string.length = 10
+      parallelism = ${dynamic_test_parallelism}
+      schema = {
+        fields {
+          c_map = "map<string, array<int>>"
+          c_array = "array<int>"
+          c_string = string
+          c_boolean = boolean
+          c_tinyint = tinyint
+          c_smallint = smallint
+          c_int = int
+          c_bigint = bigint
+          c_float = float
+          c_double = double
+          c_decimal = "decimal(30, 8)"
+          c_null = "null"
+          c_bytes = bytes
+          c_date = date
+          c_timestamp = timestamp
+          c_row = {
+            c_map = "map<string, map<string, string>>"
+            c_array = "array<int>"
+            c_string = string
+            c_boolean = boolean
+            c_tinyint = tinyint
+            c_smallint = smallint
+            c_int = int
+            c_bigint = bigint
+            c_float = float
+            c_double = double
+            c_decimal = "decimal(30, 8)"
+            c_null = "null"
+            c_bytes = bytes
+            c_date = date
+            c_timestamp = timestamp
+          }
+        }
+      }
+    }
+
+    FakeSource {
+          result_table_name = table2
+          row.num = ${dynamic_test_row_num_per_parallelism}
+          split.num = 5
+          split.read-interval = 3000
+          map.size = 10
+          array.size = 10
+          bytes.length = 10
+          string.length = 10
+          parallelism = ${dynamic_test_parallelism}
+          schema = {
+            fields {
+              c_map = "map<string, array<int>>"
+              c_array = "array<int>"
+              c_string = string
+              c_boolean = boolean
+              c_tinyint = tinyint
+              c_smallint = smallint
+              c_int = int
+              c_bigint = bigint
+              c_float = float
+              c_double = double
+              c_decimal = "decimal(30, 8)"
+              c_null = "null"
+              c_bytes = bytes
+              c_date = date
+              c_timestamp = timestamp
+              c_row = {
+                c_map = "map<string, map<string, string>>"
+                c_array = "array<int>"
+                c_string = string
+                c_boolean = boolean
+                c_tinyint = tinyint
+                c_smallint = smallint
+                c_int = int
+                c_bigint = bigint
+                c_float = float
+                c_double = double
+                c_decimal = "decimal(30, 8)"
+                c_null = "null"
+                c_bytes = bytes
+                c_date = date
+                c_timestamp = timestamp
+              }
+            }
+          }
+        }
+}
+
+transform {
+}
+
+sink {
+  LocalFile {
+    source_table_name = table1
+    path="/tmp/hive/warehouse/${dynamic_test_case_name}" # dynamic_test_case_name will be replace to the final file name before test run
+    field_delimiter="\t"
+    row_delimiter="\n"
+    file_name_expression="${transactionId}_${now}"
+    file_format="text"
+    filename_time_format="yyyy.MM.dd"
+    is_enable_transaction=true
+    save_mode="error"
+    }
+
+  LocalFile {
+    source_table_name = table2
+    path="/tmp/hive/warehouse/${dynamic_test_case_name}" # dynamic_test_case_name will be replace to the final file name before test run
+    field_delimiter="\t"
+    row_delimiter="\n"
+    file_name_expression="${transactionId}_${now}"
+    file_format="text"
+    filename_time_format="yyyy.MM.dd"
+    is_enable_transaction=true
+    save_mode="error"
+    }
+}

--- a/seatunnel-engine/seatunnel-engine-core/src/main/java/org/apache/seatunnel/engine/core/parse/JobConfigParser.java
+++ b/seatunnel-engine/seatunnel-engine-core/src/main/java/org/apache/seatunnel/engine/core/parse/JobConfigParser.java
@@ -48,6 +48,7 @@ import com.hazelcast.logging.Logger;
 import lombok.Data;
 import lombok.NonNull;
 import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.lang3.StringUtils;
 import org.apache.commons.lang3.tuple.ImmutablePair;
 
 import java.net.URL;
@@ -144,10 +145,12 @@ public class JobConfigParser {
             jobConfig.getJobContext().setJobMode(EnvCommonOptions.JOB_MODE.defaultValue());
         }
 
-        if (envConfigs.hasPath(EnvCommonOptions.JOB_NAME.key())) {
-            jobConfig.setName(envConfigs.getString(EnvCommonOptions.JOB_NAME.key()));
-        } else {
-            jobConfig.setName(EnvCommonOptions.JOB_NAME.defaultValue());
+        if (StringUtils.isEmpty(jobConfig.getName())) {
+            if (envConfigs.hasPath(EnvCommonOptions.JOB_NAME.key())) {
+                jobConfig.setName(envConfigs.getString(EnvCommonOptions.JOB_NAME.key()));
+            } else {
+                jobConfig.setName(EnvCommonOptions.JOB_NAME.defaultValue());
+            }
         }
 
         if (envConfigs.hasPath(EnvCommonOptions.CHECKPOINT_INTERVAL.key())) {

--- a/seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/master/JobMaster.java
+++ b/seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/master/JobMaster.java
@@ -44,11 +44,11 @@ import org.apache.seatunnel.engine.server.dag.physical.SubPlan;
 import org.apache.seatunnel.engine.server.execution.TaskExecutionState;
 import org.apache.seatunnel.engine.server.execution.TaskGroupLocation;
 import org.apache.seatunnel.engine.server.operation.CleanTaskGroupContextOperation;
-import org.apache.seatunnel.engine.server.operation.GetTaskGroupMetricsOperation;
 import org.apache.seatunnel.engine.server.resourcemanager.ResourceManager;
 import org.apache.seatunnel.engine.server.resourcemanager.resource.SlotProfile;
 import org.apache.seatunnel.engine.server.scheduler.JobScheduler;
 import org.apache.seatunnel.engine.server.scheduler.PipelineBaseScheduler;
+import org.apache.seatunnel.engine.server.task.operation.GetTaskGroupMetricsOperation;
 
 import com.google.common.collect.Lists;
 import com.hazelcast.cluster.Address;
@@ -147,7 +147,8 @@ public class JobMaster extends Thread {
             logicalDag = nodeEngine.getSerializationService().toObject(jobImmutableInformation.getLogicalDag());
         }
 
-        CheckpointConfig checkpointConfig = mergeEnvAndEngineConfig(engineConfig.getCheckpointConfig(), jobImmutableInformation.getJobConfig().getEnvOptions());
+        CheckpointConfig checkpointConfig = mergeEnvAndEngineConfig(engineConfig.getCheckpointConfig(),
+            jobImmutableInformation.getJobConfig().getEnvOptions());
 
         final Tuple2<PhysicalPlan, Map<Integer, CheckpointPlan>> planTuple = PlanUtils.fromLogicalDAG(logicalDag,
             nodeEngine,

--- a/seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/master/JobMaster.java
+++ b/seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/master/JobMaster.java
@@ -43,11 +43,11 @@ import org.apache.seatunnel.engine.server.dag.physical.PlanUtils;
 import org.apache.seatunnel.engine.server.dag.physical.SubPlan;
 import org.apache.seatunnel.engine.server.execution.TaskExecutionState;
 import org.apache.seatunnel.engine.server.execution.TaskGroupLocation;
-import org.apache.seatunnel.engine.server.task.operation.CleanTaskGroupContextOperation;
 import org.apache.seatunnel.engine.server.resourcemanager.ResourceManager;
 import org.apache.seatunnel.engine.server.resourcemanager.resource.SlotProfile;
 import org.apache.seatunnel.engine.server.scheduler.JobScheduler;
 import org.apache.seatunnel.engine.server.scheduler.PipelineBaseScheduler;
+import org.apache.seatunnel.engine.server.task.operation.CleanTaskGroupContextOperation;
 import org.apache.seatunnel.engine.server.task.operation.GetTaskGroupMetricsOperation;
 
 import com.google.common.collect.Lists;

--- a/seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/master/JobMaster.java
+++ b/seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/master/JobMaster.java
@@ -43,7 +43,7 @@ import org.apache.seatunnel.engine.server.dag.physical.PlanUtils;
 import org.apache.seatunnel.engine.server.dag.physical.SubPlan;
 import org.apache.seatunnel.engine.server.execution.TaskExecutionState;
 import org.apache.seatunnel.engine.server.execution.TaskGroupLocation;
-import org.apache.seatunnel.engine.server.operation.CleanTaskGroupContextOperation;
+import org.apache.seatunnel.engine.server.task.operation.CleanTaskGroupContextOperation;
 import org.apache.seatunnel.engine.server.resourcemanager.ResourceManager;
 import org.apache.seatunnel.engine.server.resourcemanager.resource.SlotProfile;
 import org.apache.seatunnel.engine.server.scheduler.JobScheduler;

--- a/seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/operation/AsyncOperation.java
+++ b/seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/operation/AsyncOperation.java
@@ -26,7 +26,7 @@ import static com.hazelcast.spi.impl.operationservice.ExceptionAction.THROW_EXCE
 import org.apache.seatunnel.engine.common.exception.SeaTunnelEngineException;
 import org.apache.seatunnel.engine.common.utils.PassiveCompletableFuture;
 import org.apache.seatunnel.engine.server.SeaTunnelServer;
-import org.apache.seatunnel.engine.server.serializable.OperationDataSerializerHook;
+import org.apache.seatunnel.engine.server.serializable.ClientToServerOperationDataSerializerHook;
 
 import com.hazelcast.nio.serialization.HazelcastSerializationException;
 import com.hazelcast.nio.serialization.IdentifiedDataSerializable;
@@ -102,6 +102,6 @@ public abstract class AsyncOperation extends Operation implements IdentifiedData
 
     @Override
     public final int getFactoryId() {
-        return OperationDataSerializerHook.FACTORY_ID;
+        return ClientToServerOperationDataSerializerHook.FACTORY_ID;
     }
 }

--- a/seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/operation/CancelJobOperation.java
+++ b/seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/operation/CancelJobOperation.java
@@ -19,7 +19,7 @@ package org.apache.seatunnel.engine.server.operation;
 
 import org.apache.seatunnel.engine.common.utils.PassiveCompletableFuture;
 import org.apache.seatunnel.engine.server.SeaTunnelServer;
-import org.apache.seatunnel.engine.server.serializable.OperationDataSerializerHook;
+import org.apache.seatunnel.engine.server.serializable.ClientToServerOperationDataSerializerHook;
 
 public class CancelJobOperation extends AbstractJobAsyncOperation {
     public CancelJobOperation() {
@@ -38,6 +38,6 @@ public class CancelJobOperation extends AbstractJobAsyncOperation {
 
     @Override
     public int getClassId() {
-        return OperationDataSerializerHook.CANCEL_JOB_OPERATOR;
+        return ClientToServerOperationDataSerializerHook.CANCEL_JOB_OPERATOR;
     }
 }

--- a/seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/operation/GetJobMetricsOperation.java
+++ b/seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/operation/GetJobMetricsOperation.java
@@ -18,7 +18,7 @@
 package org.apache.seatunnel.engine.server.operation;
 
 import org.apache.seatunnel.engine.server.SeaTunnelServer;
-import org.apache.seatunnel.engine.server.serializable.OperationDataSerializerHook;
+import org.apache.seatunnel.engine.server.serializable.ClientToServerOperationDataSerializerHook;
 
 import com.hazelcast.nio.ObjectDataInput;
 import com.hazelcast.nio.ObjectDataOutput;
@@ -44,12 +44,12 @@ public class GetJobMetricsOperation extends Operation implements IdentifiedDataS
 
     @Override
     public final int getFactoryId() {
-        return OperationDataSerializerHook.FACTORY_ID;
+        return ClientToServerOperationDataSerializerHook.FACTORY_ID;
     }
 
     @Override
     public int getClassId() {
-        return OperationDataSerializerHook.GET_JOB_METRICS_OPERATOR;
+        return ClientToServerOperationDataSerializerHook.GET_JOB_METRICS_OPERATOR;
     }
 
     @Override

--- a/seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/operation/GetJobStateOperation.java
+++ b/seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/operation/GetJobStateOperation.java
@@ -19,7 +19,7 @@ package org.apache.seatunnel.engine.server.operation;
 
 import org.apache.seatunnel.engine.common.exception.SeaTunnelEngineException;
 import org.apache.seatunnel.engine.server.SeaTunnelServer;
-import org.apache.seatunnel.engine.server.serializable.OperationDataSerializerHook;
+import org.apache.seatunnel.engine.server.serializable.ClientToServerOperationDataSerializerHook;
 
 import com.hazelcast.nio.ObjectDataInput;
 import com.hazelcast.nio.ObjectDataOutput;
@@ -45,12 +45,12 @@ public class GetJobStateOperation extends Operation implements IdentifiedDataSer
 
     @Override
     public final int getFactoryId() {
-        return OperationDataSerializerHook.FACTORY_ID;
+        return ClientToServerOperationDataSerializerHook.FACTORY_ID;
     }
 
     @Override
     public int getClassId() {
-        return OperationDataSerializerHook.PRINT_MESSAGE_OPERATOR;
+        return ClientToServerOperationDataSerializerHook.PRINT_MESSAGE_OPERATOR;
     }
 
     @Override

--- a/seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/operation/GetJobStatusOperation.java
+++ b/seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/operation/GetJobStatusOperation.java
@@ -19,7 +19,7 @@ package org.apache.seatunnel.engine.server.operation;
 
 import org.apache.seatunnel.engine.core.job.JobStatus;
 import org.apache.seatunnel.engine.server.SeaTunnelServer;
-import org.apache.seatunnel.engine.server.serializable.OperationDataSerializerHook;
+import org.apache.seatunnel.engine.server.serializable.ClientToServerOperationDataSerializerHook;
 
 import com.hazelcast.nio.ObjectDataInput;
 import com.hazelcast.nio.ObjectDataOutput;
@@ -45,12 +45,12 @@ public class GetJobStatusOperation extends Operation implements IdentifiedDataSe
 
     @Override
     public final int getFactoryId() {
-        return OperationDataSerializerHook.FACTORY_ID;
+        return ClientToServerOperationDataSerializerHook.FACTORY_ID;
     }
 
     @Override
     public int getClassId() {
-        return OperationDataSerializerHook.GET_JOB_STATUS_OPERATOR;
+        return ClientToServerOperationDataSerializerHook.GET_JOB_STATUS_OPERATOR;
     }
 
     @Override

--- a/seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/operation/PrintMessageOperation.java
+++ b/seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/operation/PrintMessageOperation.java
@@ -18,7 +18,7 @@
 package org.apache.seatunnel.engine.server.operation;
 
 import org.apache.seatunnel.engine.server.SeaTunnelServer;
-import org.apache.seatunnel.engine.server.serializable.OperationDataSerializerHook;
+import org.apache.seatunnel.engine.server.serializable.ClientToServerOperationDataSerializerHook;
 
 import com.hazelcast.nio.ObjectDataInput;
 import com.hazelcast.nio.ObjectDataOutput;
@@ -42,12 +42,12 @@ public class PrintMessageOperation extends Operation implements IdentifiedDataSe
 
     @Override
     public final int getFactoryId() {
-        return OperationDataSerializerHook.FACTORY_ID;
+        return ClientToServerOperationDataSerializerHook.FACTORY_ID;
     }
 
     @Override
     public int getClassId() {
-        return OperationDataSerializerHook.PRINT_MESSAGE_OPERATOR;
+        return ClientToServerOperationDataSerializerHook.PRINT_MESSAGE_OPERATOR;
     }
 
     @Override

--- a/seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/operation/SubmitJobOperation.java
+++ b/seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/operation/SubmitJobOperation.java
@@ -19,7 +19,7 @@ package org.apache.seatunnel.engine.server.operation;
 
 import org.apache.seatunnel.engine.common.utils.PassiveCompletableFuture;
 import org.apache.seatunnel.engine.server.SeaTunnelServer;
-import org.apache.seatunnel.engine.server.serializable.OperationDataSerializerHook;
+import org.apache.seatunnel.engine.server.serializable.ClientToServerOperationDataSerializerHook;
 
 import com.hazelcast.internal.nio.IOUtil;
 import com.hazelcast.internal.serialization.Data;
@@ -42,7 +42,7 @@ public class SubmitJobOperation extends AbstractJobAsyncOperation {
 
     @Override
     public int getClassId() {
-        return OperationDataSerializerHook.SUBMIT_OPERATOR;
+        return ClientToServerOperationDataSerializerHook.SUBMIT_OPERATOR;
     }
 
     @Override

--- a/seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/operation/WaitForJobCompleteOperation.java
+++ b/seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/operation/WaitForJobCompleteOperation.java
@@ -19,7 +19,7 @@ package org.apache.seatunnel.engine.server.operation;
 
 import org.apache.seatunnel.engine.common.utils.PassiveCompletableFuture;
 import org.apache.seatunnel.engine.server.SeaTunnelServer;
-import org.apache.seatunnel.engine.server.serializable.OperationDataSerializerHook;
+import org.apache.seatunnel.engine.server.serializable.ClientToServerOperationDataSerializerHook;
 
 public class WaitForJobCompleteOperation extends AbstractJobAsyncOperation {
 
@@ -39,6 +39,6 @@ public class WaitForJobCompleteOperation extends AbstractJobAsyncOperation {
 
     @Override
     public int getClassId() {
-        return OperationDataSerializerHook.WAIT_FORM_JOB_COMPLETE_OPERATOR;
+        return ClientToServerOperationDataSerializerHook.WAIT_FORM_JOB_COMPLETE_OPERATOR;
     }
 }

--- a/seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/serializable/ClientToServerOperationDataSerializerHook.java
+++ b/seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/serializable/ClientToServerOperationDataSerializerHook.java
@@ -20,6 +20,7 @@ package org.apache.seatunnel.engine.server.serializable;
 import org.apache.seatunnel.engine.common.serializeable.SeaTunnelFactoryIdConstant;
 import org.apache.seatunnel.engine.server.operation.CancelJobOperation;
 import org.apache.seatunnel.engine.server.operation.GetJobMetricsOperation;
+import org.apache.seatunnel.engine.server.operation.GetJobStateOperation;
 import org.apache.seatunnel.engine.server.operation.GetJobStatusOperation;
 import org.apache.seatunnel.engine.server.operation.PrintMessageOperation;
 import org.apache.seatunnel.engine.server.operation.SubmitJobOperation;
@@ -37,7 +38,7 @@ import com.hazelcast.spi.annotation.PrivateApi;
  * All about the Operation's data serializable define in this class.
  */
 @PrivateApi
-public final class OperationDataSerializerHook implements DataSerializerHook {
+public final class ClientToServerOperationDataSerializerHook implements DataSerializerHook {
     public static final int PRINT_MESSAGE_OPERATOR = 0;
     public static final int SUBMIT_OPERATOR = 1;
 
@@ -48,6 +49,8 @@ public final class OperationDataSerializerHook implements DataSerializerHook {
     public static final int GET_JOB_STATUS_OPERATOR = 4;
 
     public static final int GET_JOB_METRICS_OPERATOR = 5;
+
+    public static final int GET_JOB_STATE_OPERATION = 6;
 
     public static final int FACTORY_ID = FactoryIdHelper.getFactoryId(
         SeaTunnelFactoryIdConstant.SEATUNNEL_OPERATION_DATA_SERIALIZER_FACTORY,
@@ -81,6 +84,8 @@ public final class OperationDataSerializerHook implements DataSerializerHook {
                     return new GetJobStatusOperation();
                 case GET_JOB_METRICS_OPERATOR:
                     return new GetJobMetricsOperation();
+                case GET_JOB_STATE_OPERATION:
+                    return new GetJobStateOperation();
                 default:
                     throw new IllegalArgumentException("Unknown type id " + typeId);
             }

--- a/seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/serializable/OperationDataSerializerHook.java
+++ b/seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/serializable/OperationDataSerializerHook.java
@@ -24,7 +24,6 @@ import org.apache.seatunnel.engine.server.operation.GetJobStatusOperation;
 import org.apache.seatunnel.engine.server.operation.PrintMessageOperation;
 import org.apache.seatunnel.engine.server.operation.SubmitJobOperation;
 import org.apache.seatunnel.engine.server.operation.WaitForJobCompleteOperation;
-import org.apache.seatunnel.engine.server.task.operation.DeployTaskOperation;
 
 import com.hazelcast.internal.serialization.DataSerializerHook;
 import com.hazelcast.internal.serialization.impl.FactoryIdHelper;
@@ -42,15 +41,13 @@ public final class OperationDataSerializerHook implements DataSerializerHook {
     public static final int PRINT_MESSAGE_OPERATOR = 0;
     public static final int SUBMIT_OPERATOR = 1;
 
-    public static final int DEPLOY_TASK_OPERATOR = 2;
+    public static final int WAIT_FORM_JOB_COMPLETE_OPERATOR = 2;
 
-    public static final int WAIT_FORM_JOB_COMPLETE_OPERATOR = 3;
+    public static final int CANCEL_JOB_OPERATOR = 3;
 
-    public static final int CANCEL_JOB_OPERATOR = 4;
+    public static final int GET_JOB_STATUS_OPERATOR = 4;
 
-    public static final int GET_JOB_STATUS_OPERATOR = 5;
-
-    public static final int GET_JOB_METRICS_OPERATOR = 6;
+    public static final int GET_JOB_METRICS_OPERATOR = 5;
 
     public static final int FACTORY_ID = FactoryIdHelper.getFactoryId(
         SeaTunnelFactoryIdConstant.SEATUNNEL_OPERATION_DATA_SERIALIZER_FACTORY,
@@ -76,8 +73,6 @@ public final class OperationDataSerializerHook implements DataSerializerHook {
                     return new PrintMessageOperation();
                 case SUBMIT_OPERATOR:
                     return new SubmitJobOperation();
-                case DEPLOY_TASK_OPERATOR:
-                    return new DeployTaskOperation();
                 case WAIT_FORM_JOB_COMPLETE_OPERATOR:
                     return new WaitForJobCompleteOperation();
                 case CANCEL_JOB_OPERATOR:

--- a/seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/serializable/TaskDataSerializerHook.java
+++ b/seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/serializable/TaskDataSerializerHook.java
@@ -22,6 +22,7 @@ import org.apache.seatunnel.engine.server.execution.TaskLocation;
 import org.apache.seatunnel.engine.server.task.Progress;
 import org.apache.seatunnel.engine.server.task.TaskGroupImmutableInformation;
 import org.apache.seatunnel.engine.server.task.operation.CancelTaskOperation;
+import org.apache.seatunnel.engine.server.task.operation.CleanTaskGroupContextOperation;
 import org.apache.seatunnel.engine.server.task.operation.DeployTaskOperation;
 import org.apache.seatunnel.engine.server.task.operation.GetTaskGroupAddressOperation;
 import org.apache.seatunnel.engine.server.task.operation.GetTaskGroupMetricsOperation;
@@ -80,6 +81,8 @@ public class TaskDataSerializerHook implements DataSerializerHook {
 
     public static final int GET_TASKGROUP_METRICS_OPERATION = 18;
 
+    public static final int CLEAN_TASKGROUP_CONTEXT_OPERATION = 19;
+
     public static final int FACTORY_ID = FactoryIdHelper.getFactoryId(
         SeaTunnelFactoryIdConstant.SEATUNNEL_TASK_DATA_SERIALIZER_FACTORY,
         SeaTunnelFactoryIdConstant.SEATUNNEL_TASK_DATA_SERIALIZER_FACTORY_ID
@@ -136,6 +139,8 @@ public class TaskDataSerializerHook implements DataSerializerHook {
                     return new LastCheckpointNotifyOperation();
                 case GET_TASKGROUP_METRICS_OPERATION:
                     return new GetTaskGroupMetricsOperation();
+                case CLEAN_TASKGROUP_CONTEXT_OPERATION:
+                    return new CleanTaskGroupContextOperation();
                 default:
                     throw new IllegalArgumentException("Unknown type id " + typeId);
             }

--- a/seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/serializable/TaskDataSerializerHook.java
+++ b/seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/serializable/TaskDataSerializerHook.java
@@ -24,6 +24,7 @@ import org.apache.seatunnel.engine.server.task.TaskGroupImmutableInformation;
 import org.apache.seatunnel.engine.server.task.operation.CancelTaskOperation;
 import org.apache.seatunnel.engine.server.task.operation.DeployTaskOperation;
 import org.apache.seatunnel.engine.server.task.operation.GetTaskGroupAddressOperation;
+import org.apache.seatunnel.engine.server.task.operation.GetTaskGroupMetricsOperation;
 import org.apache.seatunnel.engine.server.task.operation.NotifyTaskStatusOperation;
 import org.apache.seatunnel.engine.server.task.operation.checkpoint.BarrierFlowOperation;
 import org.apache.seatunnel.engine.server.task.operation.checkpoint.CloseRequestOperation;
@@ -52,7 +53,9 @@ public class TaskDataSerializerHook implements DataSerializerHook {
     public static final int TASK_GROUP_INFO_TYPE = 4;
 
     public static final int SOURCE_UNREGISTER_TYPE = 5;
+
     public static final int GET_TASKGROUP_ADDRESS_TYPE = 6;
+
     public static final int SINK_REGISTER_TYPE = 7;
 
     public static final int SINK_PREPARE_COMMIT_TYPE = 8;
@@ -74,6 +77,8 @@ public class TaskDataSerializerHook implements DataSerializerHook {
     public static final int BARRIER_FLOW_OPERATOR = 16;
 
     public static final int LAST_CHECKPOINT_NOTIFY = 17;
+
+    public static final int GET_TASKGROUP_METRICS_OPERATION = 18;
 
     public static final int FACTORY_ID = FactoryIdHelper.getFactoryId(
         SeaTunnelFactoryIdConstant.SEATUNNEL_TASK_DATA_SERIALIZER_FACTORY,
@@ -129,6 +134,8 @@ public class TaskDataSerializerHook implements DataSerializerHook {
                     return new BarrierFlowOperation();
                 case LAST_CHECKPOINT_NOTIFY:
                     return new LastCheckpointNotifyOperation();
+                case GET_TASKGROUP_METRICS_OPERATION:
+                    return new GetTaskGroupMetricsOperation();
                 default:
                     throw new IllegalArgumentException("Unknown type id " + typeId);
             }

--- a/seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/task/operation/CleanTaskGroupContextOperation.java
+++ b/seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/task/operation/CleanTaskGroupContextOperation.java
@@ -15,16 +15,25 @@
  * limitations under the License.
  */
 
-package org.apache.seatunnel.engine.server.operation;
+package org.apache.seatunnel.engine.server.task.operation;
 
 import org.apache.seatunnel.engine.server.SeaTunnelServer;
 import org.apache.seatunnel.engine.server.execution.TaskGroupLocation;
+import org.apache.seatunnel.engine.server.serializable.TaskDataSerializerHook;
 
+import com.hazelcast.nio.ObjectDataInput;
+import com.hazelcast.nio.ObjectDataOutput;
+import com.hazelcast.nio.serialization.IdentifiedDataSerializable;
 import com.hazelcast.spi.impl.operationservice.Operation;
 
-public class CleanTaskGroupContextOperation extends Operation {
+import java.io.IOException;
+
+public class CleanTaskGroupContextOperation extends Operation implements IdentifiedDataSerializable {
 
     private TaskGroupLocation taskGroupLocation;
+
+    public CleanTaskGroupContextOperation() {
+    }
 
     public CleanTaskGroupContextOperation(TaskGroupLocation taskGroupLocation) {
         this.taskGroupLocation = taskGroupLocation;
@@ -37,5 +46,27 @@ public class CleanTaskGroupContextOperation extends Operation {
         SeaTunnelServer service = getService();
         service.getTaskExecutionService().notifyCleanTaskGroupContext(taskGroupLocation);
 
+    }
+
+    @Override
+    public int getFactoryId() {
+        return TaskDataSerializerHook.FACTORY_ID;
+    }
+
+    @Override
+    public int getClassId() {
+        return TaskDataSerializerHook.CLEAN_TASKGROUP_CONTEXT_OPERATION;
+    }
+
+    @Override
+    protected void writeInternal(ObjectDataOutput out) throws IOException {
+        super.writeInternal(out);
+        out.writeObject(taskGroupLocation);
+    }
+
+    @Override
+    protected void readInternal(ObjectDataInput in) throws IOException {
+        super.readInternal(in);
+        taskGroupLocation = in.readObject();
     }
 }

--- a/seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/task/operation/GetTaskGroupMetricsOperation.java
+++ b/seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/task/operation/GetTaskGroupMetricsOperation.java
@@ -15,21 +15,30 @@
  * limitations under the License.
  */
 
-package org.apache.seatunnel.engine.server.operation;
+package org.apache.seatunnel.engine.server.task.operation;
 
 import org.apache.seatunnel.api.common.metrics.RawJobMetrics;
 import org.apache.seatunnel.engine.server.execution.TaskGroupLocation;
 import org.apache.seatunnel.engine.server.metrics.JobMetricsCollector;
+import org.apache.seatunnel.engine.server.serializable.TaskDataSerializerHook;
 
 import com.hazelcast.cluster.Address;
 import com.hazelcast.logging.ILogger;
+import com.hazelcast.nio.ObjectDataInput;
+import com.hazelcast.nio.ObjectDataOutput;
+import com.hazelcast.nio.serialization.IdentifiedDataSerializable;
 import com.hazelcast.spi.impl.NodeEngineImpl;
 import com.hazelcast.spi.impl.operationservice.Operation;
 
-public class GetTaskGroupMetricsOperation extends Operation {
+import java.io.IOException;
+
+public class GetTaskGroupMetricsOperation extends Operation implements IdentifiedDataSerializable {
 
     private TaskGroupLocation taskGroupLocation;
     private RawJobMetrics response;
+
+    public GetTaskGroupMetricsOperation() {
+    }
 
     public GetTaskGroupMetricsOperation(TaskGroupLocation taskGroupLocation) {
         this.taskGroupLocation = taskGroupLocation;
@@ -55,7 +64,29 @@ public class GetTaskGroupMetricsOperation extends Operation {
     }
 
     @Override
+    protected void writeInternal(ObjectDataOutput out) throws IOException {
+        super.writeInternal(out);
+        out.writeObject(taskGroupLocation);
+    }
+
+    @Override
+    protected void readInternal(ObjectDataInput in) throws IOException {
+        super.readInternal(in);
+        taskGroupLocation = in.readObject();
+    }
+
+    @Override
     public Object getResponse() {
         return response;
+    }
+
+    @Override
+    public int getFactoryId() {
+        return TaskDataSerializerHook.FACTORY_ID;
+    }
+
+    @Override
+    public int getClassId() {
+        return TaskDataSerializerHook.GET_TASKGROUP_METRICS_OPERATION;
     }
 }

--- a/seatunnel-engine/seatunnel-engine-server/src/main/resources/META-INF/services/com.hazelcast.DataSerializerHook
+++ b/seatunnel-engine/seatunnel-engine-server/src/main/resources/META-INF/services/com.hazelcast.DataSerializerHook
@@ -15,7 +15,7 @@
 # limitations under the License.
 #
 
-org.apache.seatunnel.engine.server.serializable.OperationDataSerializerHook
+org.apache.seatunnel.engine.server.serializable.ClientToServerOperationDataSerializerHook
 org.apache.seatunnel.engine.server.serializable.TaskDataSerializerHook
 org.apache.seatunnel.engine.server.serializable.ResourceDataSerializerHook
 org.apache.seatunnel.engine.server.serializable.CheckpointDataSerializerHook

